lua/obsidian/util.lua

```diff
@@ -266,4 +266,22 @@ util.table_length = function(x)
   return n
 end
 
+-- Determines if cursor is currently inside markdown link
+-- @return integer, integer
+util.cursor_on_markdown_link = function()
+  local current_line = vim.api.nvim_get_current_line()
+  local cur_row, cur_col = unpack(vim.api.nvim_win_get_cursor(0))
+  local current_line_lh = current_line:sub(1, cur_col)
+  
+    print(cur_row, cur_col)
+  local open = current_line_lh:find("%[%[[^%[]") 
+  local close = current_line:find("%]%]", cur_col)
```

Comment:
There's a situation where this results in a false positive. Say the current line has two links:

```
here's a link [[abc]] and here's another link [[xyz]]
```

And the cursor is somewhere in between the two links. So I think you need to do something like this:

```suggestion
  local open = current_line_lh:find("%[%[[^%[%]]*$") 
  local close = current_line:find("^[^%[%]]*%]%]", cur_col)
```

Reply:
Sounds good, I never tested my suggestion so maybe I got something wrong 😉